src/Security/LoginForm.php

```diff
@@ -15,26 +15,56 @@
  */
 abstract class LoginForm extends Form
 {
+    /**
+     * @deprecated 4.4.0:5.0.0 Use getAuthenticatorClass() or setAuthenticatorClass() instead
+     * @var string
+     */
+    protected $authenticator_class;
```

Comment:
If someone has extended `LoginForm` and access `$this->authenticator_class` directly within their new class, won't this change break their code?

Reply:
Yeah true, I guess we need to read from this prop still from the places that were using it, that way B/C would be preserved.

e.g. `HiddenField::create("AuthenticationMethod", null, $this->authenticator_class ?: $this->getAuthenticatorClass(), $this),`

Would that make sense?

If devs are setting and reading it from extended classes in their own modules or project code, it would continue to work anyway. The break would be if someone hadn't set the property but was reading from it - it's protected, so would need to be done in the same hierarchy as this class

Reply:
The only problem would be if someone relies on `$this->authenticator_class` being set by the base class but then decided to get directly from another method. I'm not sure there's a nice elegant way around this ... short of keeping the existing name.

Scenarios like this are why I think we should define more methods and properties as private.

Reply:
I've always been more for protected over private, but as long as we provide transparent getters and setters I guess there's no reason we shouldn't have private properties. Agreed that in the end this replacement would be too risky. We could deprecate the property without providing an immediate replacement, that might at least encourage people not to use the property directly?